### PR TITLE
Scalar Input: switch to MomentumOptimizer. Get tests passing.

### PR DESCRIPTION
**Pull request by jyegerlehner**
This PR is against ibab's add-scalar-input branch.

I noticed when I tried to let the test run more iterations, the loss would be dropping nicely and then suddenly pop up to a large number. It would do this for all three models, one-hot, with-biases, scalar-input. I tried adding regularization but that didn't help. Then I replaced AdamOptimizer with MomentumOptimizer and the problem vanished. Seems to be an instability with Adam?

Anyway, these tests were passing on my machine now.

### CODE REVIEW

File: test/test_model.py [hunk 1]
```diff
@@ -51,20 +55,22 @@ def testEndToEndTraining(self):
...
-        optimizer = tf.train.AdamOptimizer(learning_rate=0.02)
+        optimizer = tf.train.MomentumOptimizer(learning_rate=LEARN_RATE,
+                                               momentum=MOMENTUM)
         trainable = tf.trainable_variables()
         optim = optimizer.minimize(loss, var_list=trainable)
         init = tf.initialize_all_variables()
 
         max_allowed_loss = 0.1
-        loss_val = max_allowed_loss
+        val = max_allowed_loss
```
Comment (jyegerlehner) [inline test/test_model.py]: Oops. Don't know what happened here.

**Comment by ibab** [comment]
Ah, I never even considered that it could be caused by the optimizer!
